lib/src/rules/dartdoc_summary_line.dart

```diff
@@ -0,0 +1,181 @@
+const maxSummaryLength = 140;
+
+const _desc =
+    r'Start your DartDoc comment with a single, brief sentence that ends with a period.';
+
+const _details = r'''
+From [Effective Dart](https://dart.dev/guides/language/effective-dart/documentation#do-start-doc-comments-with-a-single-sentence-summary):
+
+**DO** start doc comments with a single-sentence summary.
+
+Start your doc comment with a brief, user-centric description ending with a
+period. A sentence fragment is often sufficient. Provide just enough context
+for the reader to orient themselves and decide if they should keep reading or
+look elsewhere for the solution to their problem.
+
+For example:
+
+**GOOD:**
+```dart
+/// Deletes the file at [path] from the file system.
+void delete(String path) {
+  ...
+}
+```
+
+**BAD:**
+```dart
+/// Depending on the state of the file system and the user's permissions,
+/// certain operations may or may not be possible. If there is no file at
+/// [path] or it can't be accessed, this function throws either [IOError]
+/// or [PermissionError], respectively. Otherwise, this deletes the file.
+void delete(String path) {
+  ...
+}
+```
+
+From [Effective Dart](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph):
+
+**DO** separate the first sentence of a doc comment into its own paragraph.
+
+Add a blank line after the first sentence to split it out into its own
+paragraph. If more than a single sentence of explanation is useful, put the
+rest in later paragraphs.
+
+This helps you write a tight first sentence that summarizes the documentation.
+Also, tools like dart doc use the first paragraph as a short summary in places
+like lists of classes and members.
+
+For example:
+
+**GOOD:**
+```dart
+/// Deletes the file at [path].
+///
+/// Throws an [IOError] if the file could not be found. Throws a
+/// [PermissionError] if the file is present but could not be deleted.
+void delete(String path) {
+  ...
+}
+```
+
+**BAD:**
+```dart
+/// Deletes the file at [path]. Throws an [IOError] if the file could not
+/// be found. Throws a [PermissionError] if the file is present but could
+/// not be deleted.
+void delete(String path) {
+  ...
+}
+```
+''';
+
+/// Enforces that DartDoc blocks start with a single, brief line that ends in a period.
+///
+/// (Like the one above!)
+class DartdocSummaryLine extends LintRule {
+  DartdocSummaryLine()
+      : super(
+            name: 'dartdoc_summary_line',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(
+      NodeLintRegistry registry, LinterContext context) {
+    var visitor = _Visitor(this);
+    registry.addComment(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor<void> {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  /**
+   * Visits a [Comment] node, and finds and validates its summary line.
```

Comment:
Why is this not written with `///` delimiters?

Reply:
I was using this file as tests for the linter itself, I can rewrite everything as "///" comments before merging (once that actual unit tests exist)

Reply:
Removed alongside the code to handle `/**` comments.